salt/docker/init.sls

```diff
@@ -14,9 +14,10 @@ include:
         Environment="HTTP_PROXY={{ salt['pillar.get']('proxy:http', '') }}"
         Environment="HTTPS_PROXY={{ salt['pillar.get']('proxy:https', '') }}"
         Environment="NO_PROXY={{ salt['pillar.get']('proxy:no_proxy', '') }}"
-  cmd.run:
-    - name: systemctl daemon-reload
-    - stateful: True
+  module.run:
+    - name: service.systemctl_reload
```

Comment:
Does `sytemctl reload` restart the `docker` daemon on a changes on this file? Looking at the man page I'm not sure: _"Reload the systemd manager configuration. This will rerun all generators (see systemd.generator(7)), reload all unit files, and recreate the entire dependency tree. While the daemon is being reloaded, all sockets systemd listens on behalf of user configuration will stay accessible."_

Reply:
No, it won't restart the daemon - all we're doing in this PR is ensuring systemd is reloaded, if we're failing to reload/restart docker on changes to systemd unit files, that's a separate bug we should file and fix!

Reply:
By the way, have you actually check this `service.systemctl_reload` works fine? We did the `cmd.run: systemctl daemon-reload` because it was not working fine in the past, but maybe it has been fixed...